scripts/ci_tests/rust_tests.sh

```diff
@@ -273,91 +367,145 @@ function stress_new_tests_check_for_flakiness() {
+    local test_only_changed_crates=${TEST_ONLY_CHANGED_CRATES:-false}
+    local changed_crates_rust=${CI_CHANGED_CRATES}
+    local changed_crates_external=${CI_CHANGED_EXTERNAL_CRATES}
+    local restart_postgres=${RESTART_POSTGRES:-false}
+
+    # check if all conditions are false and early return
+    if [ "$run_rust_tests" == "false" ] && [ "$run_external_crates" == "false" ] && [ "$run_tests_using_postgres" == "false" ] && [ "$run_move_examples_rdeps_tests" == "false" ]; then
+        echo "No conditions are set to run tests. Exiting."
         exit 1
     fi
 
+    # check if external crates are set
+    if [ "$run_external_crates" == "true" ]; then
+        local changed_crates_external_filter=$(build_filterset_changed_crates "${test_only_changed_crates}" "${changed_crates_external}")
+        local exclude_set_external=$(build_filterset_excluded "${EXCLUDE_SET_EXTERNAL[@]}")
+        local combined_set_external=$(build_filterset_combined "$changed_crates_external_filter" "$exclude_set_external")
+
+        # first run tests for external crates (they are not part of the workspace)
+        if [ "$test_type" == $TEST_TYPE_NEXTEST ]; then
+            run_cargo_nextest "$combined_set_external" ".config/nextest_external.toml" "external-crates/move/Cargo.toml"
+        elif [ "$test_type" == $TEST_TYPE_SIMTEST ]; then
+            run_cargo_simtest "$combined_set_external" "external-crates/move/Cargo.toml"
+        fi
+    fi
+
+    # check again if any of the other conditions are set, in case only external crates were set
+    if [ "$run_rust_tests" == "false" ] && [ "$run_tests_using_postgres" == "false" ] && [ "$run_move_examples_rdeps_tests" == "false" ]; then
+        exit 0
+    fi
+
+    local combined_set=$(build_filterset_tests "$run_rust_tests" "$run_tests_using_postgres" "$run_move_examples_rdeps_tests" "$test_only_changed_crates" "$changed_crates_rust")
+
+    # check if a restart of postgres is needed
+    if [ "$run_tests_using_postgres" == "true" ] && [ "$restart_postgres" == "true" ]; then
+        restart_postgres_docker
+    fi
+
+    # run tests
+    if [ "$test_type" == $TEST_TYPE_NEXTEST ]; then
+        run_cargo_nextest "$combined_set"
+    elif [ "$test_type" == $TEST_TYPE_SIMTEST ]; then
+        run_cargo_simtest "$combined_set"
+    fi
+
+    if [ "$test_type" == $TEST_TYPE_NEXTEST ] && [ "$run_tests_using_postgres" == "true" ]; then
+        # Iota-indexer's RPC tests, which depend on a shared runtime, are incompatible with nextest due to its process-per-test execution model.
+        # cargo test, on the other hand, allows tests to share state and resources by default.
+        print_and_run_command "cargo test --profile simulator --package iota-indexer --test rpc-tests --all-features ${ENABLE_NO_CAPTURE:+--nocapture}"
+    fi
+}
+
+function run_tests() {
+    filter_and_run_tests $TEST_TYPE_NEXTEST
+}
+
+function run_simtests() {
```

Comment:
there's now both `run_tests()` & `rust_crates()`

and `run_simtests()` / `simtests()`

Can one of each be eliminated ?

Reply:
`rust_crates` and `simtests` do exactly what they did before, that's why I left them there.
This is the entry point to run a subset of tests locally.

The new `run_tests` and `run_simtests` can do everything, but need to be configured via environmental vars, which are set by the CI.